src/io/flutter/survey/FlutterSurveyService.java

```diff
@@ -9,68 +9,56 @@
 import com.google.gson.JsonParser;
 import com.google.gson.JsonSyntaxException;
 import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.application.ApplicationManager;
 import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.util.io.HttpRequests;
 import org.jetbrains.annotations.Nullable;
 
-import java.io.BufferedReader;
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.net.MalformedURLException;
-import java.net.URL;
-import java.nio.charset.StandardCharsets;
 import java.util.concurrent.TimeUnit;
-import java.util.stream.Collectors;
 
 public class FlutterSurveyService {
   private static final String FLUTTER_LAST_SURVEY_CONTENT_CHECK_KEY = "FLUTTER_LAST_SURVEY_CONTENT_CHECK_KEY";
   private static final long CHECK_INTERVAL_IN_MS = TimeUnit.HOURS.toMillis(40);
+  private static final String CONTENT_URL = "https://flutter.dev/f/flutter-survey-metadata.json";
   private static final PropertiesComponent properties = PropertiesComponent.getInstance();
   private static final Logger LOG = Logger.getInstance(FlutterSurveyService.class);
-  private static URL CONTENT_URL;
   private static FlutterSurvey cachedSurvey;
 
-  static {
-    try {
-      CONTENT_URL = new URL("https://flutter.dev/f/flutter-survey-metadata.json");
-    }
-    catch (MalformedURLException e) {
-      // Shouldn't happen.
-    }
-  }
 
   private static boolean timeToUpdateCachedContent() {
     // Don't check more often than once a day.
+    final long currentTimeMillis = System.currentTimeMillis();
     final long lastCheckedMillis = properties.getOrInitLong(FLUTTER_LAST_SURVEY_CONTENT_CHECK_KEY, 0);
-    return System.currentTimeMillis() - lastCheckedMillis >= CHECK_INTERVAL_IN_MS;
+    final boolean timeToUpdateCache = currentTimeMillis - lastCheckedMillis >= CHECK_INTERVAL_IN_MS;
+    if (timeToUpdateCache) {
+      properties.setValue(FLUTTER_LAST_SURVEY_CONTENT_CHECK_KEY, String.valueOf(currentTimeMillis));
+      return true;
+    }
+    return false;
   }
 
+  @Nullable
   static FlutterSurvey getLatestSurveyContent() {
-    if (timeToUpdateCachedContent() || cachedSurvey == null) {
-      cachedSurvey = fetchSurveyContent();
+    // Update the cache daily.
+    if (timeToUpdateCachedContent()) {
+      ApplicationManager.getApplication().executeOnPooledThread(() -> {
+        cachedSurvey = fetchSurveyContent();
+      });
     }
     return cachedSurvey;
```

Comment:
Don't you need to wait for the Future on line 45 to complete?

Reply:
No.  I've opted to make this async.  The assigned `cachedSurvey` will get picked up on a future editor tab switch or file open.  Precise timing isn't important.  (But not blocking **_is_**.)

Reply:
Got it. That might be worth explaining in the code.

Reply:
Good idea.  Comment added.